xoto3/dynamodb/utils/expressions.py

```diff
@@ -27,20 +27,25 @@ def make_unique_expr_attr_key(
     raise RuntimeError("Went off the end of an infinite generator")
 
 
+def validate_attr_key(attr_name: str):
+    if _filter_alphanum(attr_name) != attr_name:
+        raise ValueError(f"Attribute name contains invalid characters: '{attr_name}'")
+
+
 def add_variables_to_expression(query_dict: dict, variables: dict) -> dict:
     """Attempt to make it easier to develop a query"""
     ea_names = query_dict.get("ExpressionAttributeNames", {})
     ea_values = query_dict.get("ExpressionAttributeValues", {})
     for k, v in variables.items():
-        ea_key = make_unique_expr_attr_key(k, set(ea_names) | set(ea_values))
-        name = f"#{ea_key}"
+        validate_attr_key(k)
```

Comment:
`make_unique_expr_attr_key` cannot be used here as there's no clean way to communicate back to the consumer of `add_varaibles_to_expression` that a replacement value was generated and used. Since that consumer is responsible for creating the actual query statement using the generated placeholders, that's either going to be confusing...or a bug-producing problem.

In the case where, say, a query which already contains a partially constructed set of placeholders (say, with an orgID filter already in place), if the method was invoked to add an orgID filter, that would result in a 2nd orgID set of placeholders (`orgID0`). The calling code, however, would still likely reference the original `orgID` placeholder, not knowing an additional was created. The 2nd orgID filter would be completely ignored. AWS will _probably_ blow up if an unused placeholder exists, but I would rather this blow up immediately rather than relying on AWS to take issue with the query.

Rather than calling the code to create a unique key, I am instead validating that the supplied key complies with the alphanumeric constraint.